jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -307,15 +307,20 @@ public Response call(String httpMethod, URL url, Request request) throws IOExcep
     Preconditions.checkArgument(isHttpsProtocol(url));
     switch (failoverHistory.getOrDefault(url.getHost() + ":" + url.getPort(), Failover.NONE)) {
       case HTTP:
-        return Optional.of(call(httpMethod, toHttp(url), request, getHttpTransport(true)));
+        return Optional.of(call(httpMethod, toHttp(url), request, getHttpTransport(true), true));
       case INSECURE_HTTPS:
-        return Optional.of(call(httpMethod, url, request, getHttpTransport(false)));
+        return Optional.of(call(httpMethod, url, request, getHttpTransport(false), true));
```

Comment:
If in the history, then the protocol is already fixed for the server. (That is, we've gone through failover before and the result is cached.) These are terminal calls, so enable retrying.